### PR TITLE
[Feature] accessibility for app information and settings (closes #1127)

### LINKED ISSUE #1127: "Active Days" go down or are stuck
## UPDATE 

This will be fixed with version 1.15, which will change the active days to show the days since the installation of the app.
See #2132

## Avoid duplicates

* [x] This issue has not already been raised before
* [x] If you are proposing a new feature, please do so in [CWA-Wishlist](https://github.com/corona-warn-app/cwa-wishlist)

## Current Implementation
https://github.com/corona-warn-app/cwa-app-ios/blob/11e3821285b0ca0cc46310f0d6d2d747f907c4d7/src/xcode/ENA/ENA/Source/Workers/TracingStatusHistory.swift#L94

So out of this code and from [this](https://github.com/corona-warn-app/cwa-documentation/issues/389#issuecomment-667577152) and [this](https://github.com/corona-warn-app/cwa-documentation/issues/412#issuecomment-686388303) statement from @tkowark and @thomasaugsten, I understand the counting happens like this:

1. App counts seconds Exposure Notification was active.
2. If the counted number equals 86400sec this is shown as one day to the user. (I think the Number can be lower also, it gets rounded in some way, but I don't know how)
3. After 336 * 3600s (2 weeks) the App shows: "Risiko-Ermittlung dauerhaft aktiv"

**If this assumption is wrong/incomplete could you please explain how it works (and mainly how the rounding works)?**

## Suggested Enhancement
With the current Implementation there is the problem that when the user deactivates "Exposure Notification" during the night (activate flight mode, deactivate Bluetooth, etc.) this time is also deducted from the "active days". 
Because of this is is possible that you are stuck at one day or that the active Days go down although you had "Exposure Notification" active for the whole day, but not the night. 

I don't (yet) have a concrete proposal how this could be solved...

## Expected Benefits
It is very confusing for the user to see the Active Days go down again or stay the same. Most users won't know where the problem is and just think that the App did not work right in the last days. (Because they didn't do anything wrong, they just switched off Bluetooth during the night). 
So the expected Benefit is that the users don't get confused anymore.

---

## Related Issues
- https://github.com/corona-warn-app/cwa-documentation/issues/412
- https://github.com/corona-warn-app/cwa-wishlist/issues/102
- https://github.com/corona-warn-app/cwa-app-ios/issues/1739#issue-774547725
---
Internal Tracking ID: [EXPOSUREAPP-2556](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-2556)

### PR DESCRIPTION
**Pull request by i336616**
https://jira.itc.sap.com/browse/EXPOSUREAPP-1127

Adds accessibilityLabels to the images on the App Information screens
Note: some of the screens have not been provided with accessibility content for the images:
- Über die App
- Datenshutz

## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [ ] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift [hunk 1]
```diff
@@ -94,4 +94,9 @@ extension AppInformationViewController {
 			self.execute(action: action)
 		}
 	}
+
+	func setupAccessibility() {
+
+	}
+
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift:101]: Why is this empty? Can it be deleted?.

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift [hunk 2]
```diff
@@ -48,101 +48,111 @@ private extension DynamicAction {
...
+				.body(text: AppStrings.AppInformation.imprintSection4Text)
 			]
 		)
 	])
 
 	private static let privacyModel = DynamicTableViewModel([
 		.section(
-			header: .image(UIImage(named: "Illu_Appinfo_Datenschutz"), height: 230),
+			header: .image(UIImage(named: "Illu_Appinfo_Datenschutz"),
+						   //accessibilityLabel: AppStrings.AppInformation.privacyImageDescription,
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:132]: Why is his commented out?

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift [hunk 3]
```diff
@@ -48,101 +48,111 @@ private extension DynamicAction {
...
-				.body(text: "App_Information_Imprint_Section2_Text".localized),
-				// .headline(text: "App_Information_Legal_Section3_Title".localized),
-				.body(text: "App_Information_Imprint_Section3_Text".localized),
-				.headline(text: "App_Information_Imprint_Section4_Title".localized),
-				.body(text: "App_Information_Imprint_Section4_Text".localized)
+				.headline(text: AppStrings.AppInformation.imprintSection1Title),
+				.body(text: AppStrings.AppInformation.imprintSection1Text),
+				.headline(text: AppStrings.AppInformation.imprintSection2Title),
+				.body(text: AppStrings.AppInformation.imprintSection2Text),
+				// .headline(text: AppStrings.AppInformation.legalSection3Title),
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:121]: What will collide with #258. Be careful when merging.

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift [hunk 4]
```diff
@@ -48,101 +48,111 @@ private extension DynamicAction {
...
 		)
 	]
 }
 
 extension AppInformationViewController {
 	private static let aboutModel = DynamicTableViewModel([
 		.section(
-			header: .image(UIImage(named: "Illu_AppInfo_UeberApp"), height: 230),
+			header: .image(UIImage(named: "Illu_AppInfo_UeberApp"),
+						   //accessibilityLabel: AppStrings.AppInformation.aboutImageDescription,
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:85]: Why is this commented out?

File: src/xcode/ENA/ENA/Source/Models/AppInformation/AppInformationDetailModelData.swift [hunk 5]
```diff
@@ -0,0 +1,100 @@
+// Corona-Warn-App
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Models/AppInformation/AppInformationDetailModelData.swift:1]: This file was deleted and should not be re-added again.

File: src/xcode/ENA/ENA/Source/Models/AppInformation/AppInformationDetailModel.swift [hunk 6]
```diff
@@ -0,0 +1,38 @@
+// Corona-Warn-App
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Models/AppInformation/AppInformationDetailModel.swift:1]: This file was deleted and should not be re-added again.

File: src/xcode/ENA/ENA/Source/AppDelegate.swift [hunk 7]
```diff
@@ -339,6 +339,11 @@ extension AppDelegate: ENATaskExecutionDelegate {
 			return
 		}
 
+		self.taskScheduler.notificationManager.presentNotification(
+			title: "BGTask event received",
+			body: "\(ENATaskIdentifier.detectExposures.backgroundTaskSchedulerIdentifier)",
+			identifier: ENATaskIdentifier.detectExposures.rawValue)
+
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/AppDelegate.swift:346]: Not really part of this PR?

File: src/xcode/ENA/ENA/Source/AppDelegate.swift [hunk 8]
```diff
@@ -377,6 +382,11 @@ extension AppDelegate: ENATaskExecutionDelegate {
 			taskScheduler.scheduleBackgroundTask(for: .fetchTestResults)
 		}
 
+		self.taskScheduler.notificationManager.presentNotification(
+			title: "BGTask event received",
+			body: "\(ENATaskIdentifier.fetchTestResults.backgroundTaskSchedulerIdentifier)",
+			identifier: ENATaskIdentifier.fetchTestResults.rawValue)
+
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/AppDelegate.swift:389]: Not really part of this PR?

**Review by eisvogel — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift (hunk 1 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift:101]: Yes, deleting it in the next commit!

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift (hunk 2 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:132]: There should be text for this image, but it's not on the JIRA. I've kept this in but commented for when they supply the accessibility text for it.

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift (hunk 4 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:85]: As above, there should be text for this image, but it's not on the JIRA. I've kept this in but commented for when they supply the accessibility text for it.

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift (hunk 2 above)
Comment (EisvogelSAP) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:132]: Should be tagged with TODO, I guess, so that we don't forget about it.

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift (hunk 4 above)
Comment (EisvogelSAP) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:85]: Should be tagged with TODO, I guess, so that we don't forget about it.

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift (hunk 3 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:121]: Updated, will wait to commit until #258 is merged

Comment (EisvogelSAP) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:121]: Should have been merged.

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift (hunk 4 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:100]: Added the TODO for this

File: src/xcode/ENA/ENA/Source/Models/AppInformation/AppInformationDetailModel.swift (hunk 6 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Source/Models/AppInformation/AppInformationDetailModel.swift]: Also removed

File: src/xcode/ENA/ENA/Source/AppDelegate.swift (hunk 7 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Source/AppDelegate.swift]: Shouldn't have been committed, was used for debugging the BGTasks

File: src/xcode/ENA/ENA/Source/AppDelegate.swift (hunk 8 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Source/AppDelegate.swift]: Shouldn't have been committed, was used for debugging the BGTasks

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift [hunk 9]
```diff
@@ -62,101 +62,113 @@ private extension DynamicAction {
...
 		.section(
-			header: .image(UIImage(named: "Illu_Appinfo_Kontakt"), height: 230),
+			header: .image(UIImage(named: "Illu_Appinfo_Kontakt"),
+						   accessibilityLabel: AppStrings.AppInformation.contactImageDescription,
+						   height: 230),
 			cells: [
-				.title2(text: "App_Information_Contact_Title".localized),
+				.title2(text: AppStrings.AppInformation.contactTitle),
 				.body(text: "App_Information_Contact_Description".localized),
 				.headline(text: "App_Information_Contact_Hotline_Title".localized),
```
Comment (EisvogelSAP) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController+DynamicTableViewModel.swift:118]: You missed these two?

**Review by eisvogel — APPROVED** [review]

File: src/xcode/ENA/ENA/Resources/Localization/en.lproj/Localizable.strings [hunk 10]
```diff
@@ -726,6 +741,11 @@
 /* Local Notifications */
 "local_notifications_viewResults" = "Display Results";
 
+// App Information - Legal
```
Comment (llanghammer-sap) [inline src/xcode/ENA/ENA/Resources/Localization/en.lproj/Localizable.strings:744]: Aren't we supposed to not change the english translations files anymore? Just making sure to be on the safe side.

And I think we should also use /* */ in the english file as well.

**Review by llanghammer-sap — CHANGES_REQUESTED** [review]
Looks good, I just have one small question. 👍

File: src/xcode/ENA/ENA/Resources/Localization/en.lproj/Localizable.strings (hunk 10 above)
Comment (i336616) [reply src/xcode/ENA/ENA/Resources/Localization/en.lproj/Localizable.strings:744]: I'll revert these in the next commit

**Review by llanghammer-sap — APPROVED** [review]
